src/lamp_py/ingestion/convert_gtfs_rt.py

```diff
@@ -353,38 +343,151 @@ def gz_to_pyarrow(
+    def continuous_pq_update(self, table: pyarrow.Table) -> None:
+        """
+        Continuous updating of local parquet files that are synced with S3
+        """
+        log = ProcessLogger("continuous_pq_update")
+        log.log_start()
+        try:
+            self.verify_table(table)
+
+            year = pc.unique(table.column("year")).to_pylist()[0]
+            month = pc.unique(table.column("month")).to_pylist()[0]
+            day = pc.unique(table.column("day")).to_pylist()[0]
+            hour = pc.unique(table.column("hour")).to_pylist()[0]
+
+            part_folder = os.path.join(
+                DEFAULT_S3_PREFIX,
+                str(self.config_type),
+                f"year={year}",
+                f"month={month}",
+                f"day={day}",
+                f"hour={hour}",
             )
-            sort_log.log_start()
-            if self.detail.table_sort_order is not None:
-                table = table.sort_by(self.detail.table_sort_order)
-            sort_log.log_complete()
-
-            write_parquet_file(
-                table=table,
-                file_type=s3_prefix,
-                s3_dir=os.path.join(
-                    os.environ["SPRINGBOARD_BUCKET"],
-                    DEFAULT_S3_PREFIX,
-                    s3_prefix,
-                ),
-                partition_cols=["year", "month", "day", "hour"],
-                visitor_func=self.send_metadata,
+            local_path = self.sync_with_s3(part_folder)
+
+            log.add_metadata(part_folder=part_folder, local_path=local_path)
+
+            # no existing table file, write new file
+            if local_path == "":
+                local_path = os.path.join(
+                    self.tmp_folder,
+                    part_folder,
+                    f"{datetime(year, month, day, hour).isoformat()}.parquet",
+                )
+                pq.write_table(table, local_path)
+
+            # join existing table file with new table data
+            else:
+                new_table_path = os.path.join(
+                    self.tmp_folder,
+                    part_folder,
+                    "new_table.parquet",
+                )
+                pq.write_table(table, new_table_path)
+
+                join_table_path = os.path.join(
+                    self.tmp_folder,
+                    part_folder,
+                    "join_table.parquet",
+                )
+                join_ds = pd.dataset((local_path, new_table_path)).sort_by(
+                    self.detail.table_sort_order
+                )
+                with pq.ParquetWriter(
+                    join_table_path, schema=table.schema
+                ) as writer:
+                    for batch in join_ds.to_batches(
+                        batch_size=self.pq_batch_size
+                    ):
+                        writer.write_batch(batch)
+
+                os.replace(join_table_path, local_path)
+                os.remove(new_table_path)
```

Comment:
this part feels like it could be its own method. that would let you keep the R0914 pylint rule on as well.

Reply:
with these changes the code looks something like

```
try:
    y, m, d, h = extract_datetime(table)
    
    file_suffix = os.path.join(
        LAMP_S3_PREFIX,
        str(self.config_type),
        f"year={year}",
        f"month={month}",
        f"day={day}",
        f"hour={hour}",
        f"{datetime(year, month, day, hour).isoformat()}.parquet",
    )
    
    if download_s3_locally(file_suffix):
        self.append_to_local_path(table, file_suffix)
    else:
        local_path = os.path.join(self.tmp_folder, file_suffix)
        pq.write_table(table, local_path)
        
    upload_path = os.path.join(os.environ["SPRINGBOARD_BUCKET"], file_suffix)
    upload_file(local_path, local_path)

except Exception as exception:
    # error handling
```

Reply:
this function has been cleaned up and the pylint exception removed.